inluxclient/client.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+// Package influxclient provides client for InfluxDB server.
+package influxclient
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+var (
+	// ErrEmptyServerURL is returned when params do have set ServerURL field
+	ErrEmptyServerURL = errors.New("empty server URL")
+)
+
+// Params holds the parameters for creating a new client.
+// The only mandatory field is ServerURL. AuthToken is also important
+// if authentication was not done outside this client.
+type Params struct {
+	// ServerURL holds the URL of the InfluxDB server to connect to.
+	// This must be non-empty. E.g. http://localhost:8086
+	ServerURL string
+
+	// AuthToken holds the authorization token for the API.
+	// This can be obtained through the GUI web browser interface.
+	AuthToken string
+
+	// DefaultTags specifies a set of tags that will be added to each written
```

Comment:
How about leaving the parameters that are strictly writer-related out for now, and adding them when the relevant functionality is implemented? That way we can be more sure that we're not leaving around API that never gets hooked up. I'd include both `BatchSize` and `FlushInterval` in this suggestion.

Reply:
Ok